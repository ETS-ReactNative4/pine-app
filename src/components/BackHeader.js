import React, { Component } from 'react';
import { StyleSheet, View, Text } from 'react-native';
import PropTypes from 'prop-types';

import getStatusBarHeight from '../utils/getStatusBarHeight';
import getNavBarHeight from '../utils/getNavBarHeight';
import headerStyles from '../styles/headerStyles';
import { withTheme } from '../contexts/theme';
import BackButton from './BackButton';

const styles = StyleSheet.create({
  header: {
    marginTop: getStatusBarHeight(),
    alignSelf: 'stretch',
    justifyContent: 'center',
    backgroundColor: 'transparent',
    height: getNavBarHeight(),
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0
  },
  title: {
    position: 'absolute',
    left: 40,
    right: 40,
    textAlign: 'center'
  }
});

class BackHeader extends Component {
  render() {
    const { showBackButton, theme } = this.props;

    return (
      <View style={styles.header}>
        { showBackButton ? <BackButton onPress={this.props.onBackPress} /> : null }

        <Text style={[headerStyles.title, theme.headerTitle, styles.title]}>
          {this.props.title}
        </Text>
      </View>
    );
  }
}

BackHeader.propTypes = {
  showBackButton: PropTypes.bool,
  onBackPress: PropTypes.func,
  title: PropTypes.string,
  theme: PropTypes.object
};

export default withTheme(BackHeader);
