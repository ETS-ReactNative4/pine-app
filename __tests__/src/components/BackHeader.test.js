import React from 'react';
import renderer from 'react-test-renderer';
import BackHeader from '../../../src/components/BackHeader';

describe('BackHeader', () => {
  it('renders correctly with showBackButton set to false', () => {
    const tree = renderer.create(
      <BackHeader showBackButton={false} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with showBackButton set to true', () => {
    const tree = renderer.create(
      <BackHeader showBackButton={true} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
