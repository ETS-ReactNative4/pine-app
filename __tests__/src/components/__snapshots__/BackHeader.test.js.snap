// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackHeader renders correctly with showBackButton set to false 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 42,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "black",
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "600",
        },
        Object {
          "left": 40,
          "position": "absolute",
          "right": 40,
          "textAlign": "center",
          "top": 36,
        },
      ]
    }
  />
</View>
`;

exports[`BackHeader renders correctly with showBackButton set to true 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 42,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 10,
        "opacity": 1,
        "position": "absolute",
        "top": 36,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#000000",
              "fontSize": 20,
              "paddingRight": 10,
            },
            undefined,
          ],
          Object {
            "fontFamily": "Entypo",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï…‰
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "black",
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "600",
        },
        Object {
          "left": 40,
          "position": "absolute",
          "right": 40,
          "textAlign": "center",
          "top": 36,
        },
      ]
    }
  />
</View>
`;
